### DISCUSSION TITLE
Clear cache for one user

### DISCUSSION THREAD

**Discussion by Fr3di** [spatie/laravel-responsecache#360] [Q&A]
I would like to clear the cache only for one user.
Example:
Each user has it's own list of items which can edit (add or remove element). Taking this actions will not be often, maybe once a few months so I would like to cache the response for some time. Now, when the user will add or remove new item, I would like to clear the cache so the user could get the newest changes right after.
The documentation says that "Logged in users will each have their own separate cache" and as far as I understand
`ResponseCache::clear();`
method clear all the cache, for all users at once.
Is there any way to clear cache only for currently logged in user?

**Comment by RChutchev** [comment]
> I would like to clear the cache only for one user. Example: Each user has it's own list of items which can edit (add or remove element). Taking this actions will not be often, maybe once a few months so I would like to cache the response for some time. Now, when the user will add or remove new item, I would like to clear the cache so the user could get the newest changes right after. The documentation says that "Logged in users will each have their own separate cache" and as far as I understand `ResponseCache::clear();` method clear all the cache, for all users at once. Is there any way to clear cache only for currently logged in user?

Yes, replace a $user->id with your property which contains the required user ID.
```
        ResponseCache::selectCachedItems()
            ->forUrls('/path')
            ->usingSuffix($user->id)
            ->forget();
```
As for me, IDK why, but ResponseCache::forget doesn't clear the cache for all users as required for us.
